### DISCUSSION TITLE
Overall vision

### DISCUSSION THREAD

**Discussion by arkrde** [matt-42/lithium#66] [General]
Hello @matt-42 ,
Thanks for creating the project. It looks very exciting. Since web frameworks can include a lot of things, would you mind sharing your vision for lithium: what you like to see in lithium, and which parts you want to delegate to other projects ?

**Comment by matt-42** [comment]
HI @arkrde ,

Sorry for the delay, I took a few weeks break on lithium to focus on other things. Here is my answer to your questions:

Lithium's mission is to make high performance web development easier and safer with C++.

It is structured as a set of libraries (a bit like boost) so each one can be learned and used separately (for example: the json serializer has to reason to be tight to the http server).

Today, it contains http APIs, DB drivers, and json serialization. For all the rest you are welcome to use any other C++ lib.

What will be added in the future is not sure yet, there has been a suggestion to add new features like websockets, Oauth2, but also to consolidate the current features with more comments, cleaner code, docs, more tests... Everything will depends on the users requests,
sponsors, and how new contributors will be able to get into the code.

Hope this answers your question,

Cheers,
Matthieu

**Reply by arkrde** [reply]
Thanks, @matt-42. This helps.
Regards,
Arnab

**Reply by unclemate** [reply]
@matt-42 I vote for websocket in the next release.
